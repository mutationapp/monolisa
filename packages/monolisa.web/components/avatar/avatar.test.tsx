import Avatar from './avatar'
import React from 'react'
import { shallow } from 'enzyme'

test.each([
  {
    processing: true,
  },
  {
    processing: false,
  },
])('Renders', ({ processing }) => {
  expect(
    shallow(<Avatar name="avatar" processing={processing} />),
  ).toMatchSnapshot()
})
