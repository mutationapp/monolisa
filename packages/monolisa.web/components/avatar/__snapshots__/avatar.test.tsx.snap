// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
<span
  className="avatar"
  data-initial=""
>
  <style
    jsx={true}
  >
    
        .avatar &gt; :global(img) {
          border-radius: 50%;
        }

        .avatar {
          width: 30px;
          height: 30px;
          line-height: 30px;
          font-size: 15px;
          display: inline-flex;
          border-radius: 50%;
          text-align: center;
          background-color: var(--foreground);
          color: var(--background);
          align-items: center;
          justify-content: center;
        }
        .avatar:after {
          content: attr(data-initial);
        }

        .processing {
          margin: 0 auto;
        }
      
  </style>
  <Spinner />
</span>
`;

exports[`Renders 2`] = `
<span
  className="avatar"
  data-initial="A"
>
  <style
    jsx={true}
  >
    
        .avatar &gt; :global(img) {
          border-radius: 50%;
        }

        .avatar {
          width: 30px;
          height: 30px;
          line-height: 30px;
          font-size: 15px;
          display: inline-flex;
          border-radius: 50%;
          text-align: center;
          background-color: var(--foreground);
          color: var(--background);
          align-items: center;
          justify-content: center;
        }
        .avatar:after {
          content: attr(data-initial);
        }

        .processing {
          margin: 0 auto;
        }
      
  </style>
</span>
`;
